components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -643,7 +643,7 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
               do k=minLevelEdgeBot(iEdge)+1,maxLevelEdgeTop(iEdge)
                  BruntVaisalaFreqTopEdge = 0.5_RKIND*(max(BruntVaisalaFreqTop(k,cell1),0.0_RKIND) + &
                                           max(BruntVaisalaFreqTop(k,cell2),0.0_RKIND))
-                 shearEdgeInv = (0.5_RKIND*(layerThickEdge(k-1,iEdge) + layerThickEdge(k,iEdge)))**2.0 &
+                 shearEdgeInv = (0.5_RKIND*(layerThickEdgeFlux(k-1,iEdge) + layerThickEdgeFlux(k,iEdge)))**2.0 &
```

Comment:
@vanroekel Suppose we are employing the upwinded `layerThicknessEdge` for momentum and tracer fluxes (here designated `layerThickEdgeFlux`). Can you weigh in on whether it is appropriate to use the central-differenced `layerThicknessEdge` (here designated `layerThickEdgeCenter`) for the GM bolus velocity computation? Here, I only retained `layerThickEdgeFlux` in the computation of the Brunt-Vaisala frequency.

Reply:
@cbegeman I think this one has to be center averaged since it is for thickness weighted averaging on edges.

I think the `BruntVaisalaFreqTopEdge` needs to be center averaged as well.  It doesn't make sense to me to use the upwind layerThickness there either.  But perhaps I'm not fully understanding your last sentence of the comment.

Reply:
@vanroekel Thanks for the prompt feedback! I'm glad we're in agreement on center averaging in general for GM.

I was thinking that perhaps we'd want to be using the upwinded `layerThicknessEdge` to compute the vertical shear when the volume fluxes through the edge are `u*h_upwind`. If that doesn't make sense, I'm happy to make the change to centered thicknesses throughout this routine.

Reply:
I see, thanks for the explanation @cbegeman.  Given this is to compute the Richardson number I think we should be consistent with the BVF edge and Shear.  Although it may be interesting to explore different options and associated sensitivities.  I definitely see your point though, but I'd like to leave centered for now until we can do testing of how sensitive the calculation is to that choice in isolation.

Reply:
@vanroekel That sounds reasonable to me. We'll go with centered thickness throughout.